src/charm.py

```diff
@@ -452,7 +466,31 @@ def _prometheus_config(self) -> str:
             "metrics_path": "/metrics",
             "honor_timestamps": True,
             "scheme": "http",
-            "static_configs": [{"targets": [f"localhost:{self._port}"]}],
+            "static_configs": [
+                {
+                    "targets": [f"localhost:{self._port}"],
+                    "labels": {
+                        "juju_model": self._topology.model,
+                        "juju_model_uuid": self._topology.model_uuid,
+                        "juju_application": self._topology.application,
+                        "juju_unit": self._topology.charm_name,
+                        "host": "localhost",
+                    },
+                }
+            ],
+            "relabel_configs": [
```

Comment:
Add comment for the purpose of relabeling?

Reply:
This isn't actually what it does, per se. This _replaces_ the `instance` label (normally an IP address) with the values here.

It's more like:

```python
re.sub(r'^(instance=).+', rf'\1{"_".join([labels[k] for k in ["juju_model", "juju_model_uuid", "juju_application", "juju_unit"] if source_labels[k]])}', value)
```

So:
`instance=1.2.3.4` -> `instance=model_test_dead-beef_fooapp_fooapp/0`